core/src/main/java/io/atomix/core/tree/DocumentPath.java

```diff
@@ -36,20 +35,67 @@
  */
 public class DocumentPath implements Comparable<DocumentPath> {
 
+  /**
+   * Creates a new {@code DocumentPath} from a period delimited path string.
+   *
+   * @param path path string
+   * @return {@code DocumentPath} instance
+   */
+  public static DocumentPath from(String path) {
+    if (path.equals(ROOT_PATH)) {
+      return ROOT;
+    }
+    return new DocumentPath(Arrays.asList(path.split(PATH_SEPARATOR)));
+  }
+
+  /**
+   * Creates a new {@code DocumentPath} from a list of path elements.
+   *
+   * @param elements path elements
+   * @return {@code DocumentPath} instance
+   */
+  public static DocumentPath from(String... elements) {
+    return from(Arrays.asList(elements));
+  }
+
+  /**
+   * Creates a new root {@code DocumentPath} from a list of path elements.
+   *
+   * @param elements path elements
+   * @return {@code DocumentPath} instance
+   */
+  public static DocumentPath from(List<String> elements) {
+    List<String> copy = new ArrayList<>();
+    copy.add("");
```

Comment:
what's the empty string for?

Reply:
```java
  /**
   * Root document path.
   */
  private static final String ROOT_PATH = "/";

  /**
   * Root document tree path.
   */
  public static final DocumentPath ROOT = new DocumentPath(ImmutableList.of(""));
```
do these still match?

Reply:
just read over the converation on gitter, might be nicer if we introduce a constant here, ROOT_ELEMENT, self documenting code ftw.

Reply:
Good plan